### PR TITLE
Add WTP XML format tests

### PR DESCRIPTION
**Pull request by angelozerr**
Add WTP XML format tests

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
@JessicaJHee here the PR which adapts Eclipse WTP XML format tests https://git.eclipse.org/c/sourceediting/webtools.sourceediting.git/tree/xml/tests/org.eclipse.wst.xml.core.tests/src/org/eclipse/wst/xml/core/tests/format/TestPartitionFormatterXML.java with LemMinx.

All tests failed because:

 * perhaps the settings configuration is not done correctly
 * some tests failed with error like
image
 I think we should do a fix for that.

I don't say, we must fix all tests because we have a different strategy from WTP (format with mixed content), but it can be interesting to invesitigate if we could do some improvement (to fix error tests) or perhaps adding new settings if we need to try to mimic WTP.

Here the results of tests:

image

File: org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/format/wtp/TestPartitionFormatterXML.java [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+import org.eclipse.lemminx.services.XMLLanguageService;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lemminx.settings.XMLFormattingOptions;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test from
+ * https://github.com/kingargyle/wtp-sourceediting/blob/master/tests/org.eclipse.wst.xml.core.tests/src/org/eclipse/wst/xml/core/tests/format/TestPartitionFormatterXML.java
```
Comment (fbricon) [inline org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/format/wtp/TestPartitionFormatterXML.java:15]: You should keep copyright headers from original sources

**Comment by angelozerr** [comment]
@JessicaJHee for your information here the default settings from WTP:

image

We need to find which lemminx configuration we need to do to set the same default value in WWD. Yesterday my PR with Lemminx format settings has been merged https://github.com/eclipse/wildwebdeveloper/pull/1009 but it requires some improvement to set by default the same format settings than WTP.

File: org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/format/wtp/TestPartitionFormatterXML.java (hunk 1 above)
Comment (angelozerr) [reply org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/format/wtp/TestPartitionFormatterXML.java]: fixed

**Comment by angelozerr** [comment]
I disabled all tests which failed. I think it can be a good idea to merge this PR and fix with several PRs the formatter to try to mimic WTP and re-enable it again the tests fixed by the futures PRs.

@fbricon are you OK to merge this PR?

**Comment by fbricon** [comment]
> @fbricon are you OK to merge this PR?

Once the build passes, yes

**Comment by fbricon** [comment]
Still

> FilePathCompletionTest.testFilePathCompletionForEmptyDoctype:274->testCompletionFor:317 expected: <9> but was: <10>

**Pull request merged**